tgui/packages/tgui/interfaces/SecurityRecords.jsx

```diff
@@ -0,0 +1,688 @@
+  const [commentModalOpen, setCommentModalOpen] = useState(false);
+  const [newComment, setNewComment] = useState('');
+  const [viewFingerprintScanner, setViewFingerprintScanner] = useState(false); // Track fingerprint scanner view
+  const [sortConfig, setSortConfig] = useState({ key: null, direction: 'asc' });
+  const [filterText, setFilterText] = useState('');
+  const [currentPhoto, setCurrentPhoto] = useState('front'); // State to track the current photo (front or side)
+
+  useEffect(() => {
+    if (Array.isArray(records)) {
+      setRecordsArray(records);
+    }
+  }, [records]);
+
+  useEffect(() => {
+    if (selectedRecord) {
+      const updatedRecord = recordsArray.find(
+        (record) => record.id === selectedRecord.id,
+      );
+      if (updatedRecord) {
+        setSelectedRecord(updatedRecord);
+      } else {
+        goBack();
+      }
+    }
+  }, [recordsArray, selectedRecord]);
+
+  const handleSave = (value) => {
+    act('update_field', { id: selectedRecord.id, field: editField, value });
+    closeEditModal();
+  };
+
+  const handleAddComment = () => {
+    if (newComment.trim()) {
+      act('add_comment', { id: selectedRecord.id, comment: newComment });
+
+      setNewComment('');
+      closeCommentModal();
+    }
+  };
+
+  const changePhoto = () => {
+    setCurrentPhoto((prevPhoto) => (prevPhoto === 'front' ? 'side' : 'front'));
+  };
+
+  const handleUpdatePhoto = () => {
+    act('update_photo', { id: selectedRecord.id, photo_profile: currentPhoto });
+  };
+
+  const handleSort = (key) => {
+    const direction =
+      sortConfig.key === key && sortConfig.direction === 'asc' ? 'desc' : 'asc';
+    setSortConfig({ key, direction });
+
+    const sortedRecords = [...recordsArray].sort((a, b) => {
+      if (a[key] < b[key]) return direction === 'asc' ? -1 : 1;
+      if (a[key] > b[key]) return direction === 'asc' ? 1 : -1;
+      return 0;
+    });
+
+    setRecordsArray(sortedRecords);
+  };
+
+  const filteredRecords = recordsArray.filter((record) =>
+    Object.values(record).some((value) =>
+      String(value).toLowerCase().includes(filterText.toLowerCase()),
+    ),
+  );
+
+  //* Functions for handling modals state
+
+  const openEditModal = (field, value) => {
+    setEditField(field);
+    setEditValue(value);
+  };
+
+  const closeEditModal = () => {
+    setEditField(null);
+    setEditValue('');
+  };
+
+  const openCommentModal = () => {
+    setCommentModalOpen(true);
+  };
+
+  const closeCommentModal = () => {
+    setCommentModalOpen(false);
+    setNewComment('');
+  };
+
+  const cellStyle = {
+    paddingTop: '4px',
+    paddingBottom: '3px',
+    textAlign: 'center',
+  };
+
+  const boxStyle = {
+    paddingTop: '5px',
+    paddingBottom: '5px',
+  };
+
+  const sectionHeaderStyle = {
+    paddingTop: '10px',
+    paddingBottom: '10px',
+  };
+
+  const grayItalicStyle = {
+    fontSize: '0.9rem',
+    color: 'gray',
+    fontStyle: 'italic',
+  };
```

Comment:
why in line styles and not in .scss?

Reply:
Had previously problem with that, but figured it out eventually.
Moved all styles to proper scss file.